example_school_library_decorator/main.rb

```diff
@@ -0,0 +1,34 @@
+require_relative 'functions'
+
+functions = Functions.new
+
+puts 'Welcome to School Library App!'
+
+puts 'Please choose an option by entering a number:'
+puts '1 - List all books'
+puts '2 - List all people'
+puts '3 - Create a person'
+puts '4 - Create a book'
+puts '5 - Create a rental'
+puts '6 - List all rentals for a given person id'
+puts '7 - Exit'
+
+response = gets.chomp
+
+case response
+when '1'
+  functions.list_of_books
+when '2'
+  functions.list_of_people
+when '3'
+  functions.create_person
+when '4'
+  functions.create_book
+when '5'
+  functions.create_rental
+when '6'
+  functions.list_of_rentals_for_person_id
+when '7'
+  puts 'Thank you for using this app!'
+end
+puts "\n"
```

Comment:
- [ ] It's a good thing you now have the logic fully set for the app. Let's improve our app just a little bit to enhance user experience. Currently, when a user performs any action, the app exits upon completion. This is not good, we should let the user decide when to close the app after performing all the actions the user wanted to perform. Consider applying the suggestion below:
```suggestion
while response != '7'
  puts 'Please choose an option by enterin a number:'
  puts '1 - List all books'
  puts '2 - List all people'
  puts '3 - Create a person'
  puts '4 - Create a book'
  puts '5 - Create a rental'
  puts '6 - List all rentals for a given person id'
  puts '7 - Exit'
  # *Lets users choose an option.
  response = gets.chomp

  case response
  when '1'
    functions.list_of_books
  when '2'
    functions.list_of_people
  when '3'
    functions.create_person
  when '4'
    functions.create_book
  when '5'
    functions.create_rental
  when '6'
    functions.list_of_rentals_for_person_id
  when '7'
    puts 'Thank you for using this app!'
  end
  puts "\n"
end
```